app/src/main/java/org/simple/clinic/summary/addphone/AddPhoneNumberDialog.kt

```diff
@@ -0,0 +1,126 @@
+package org.simple.clinic.summary.addphone
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.fragment.app.FragmentManager
+import com.google.android.material.textfield.TextInputLayout
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import kotterknife.bindView
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.patient.PatientUuid
+import org.simple.clinic.widgets.ScreenDestroyed
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.showKeyboard
+import javax.inject.Inject
+
+class AddPhoneNumberDialog : AppCompatDialogFragment() {
+
+  companion object {
+    private const val FRAGMENT_TAG = "AddPhoneNumberDialog"
+    private const val KEY_PATIENT_UUID = "patientUuid"
+
+    fun show(patientUuid: PatientUuid, fragmentManager: FragmentManager) {
+      val existingFragment = fragmentManager.findFragmentByTag(FRAGMENT_TAG)
+
+      if (existingFragment != null) {
+        fragmentManager
+            .beginTransaction()
+            .remove(existingFragment)
+            .commitNowAllowingStateLoss()
+      }
+
+      val fragment = AddPhoneNumberDialog().apply {
+        isCancelable = false
+        arguments = Bundle(1).apply {
+          putSerializable(KEY_PATIENT_UUID, patientUuid)
+        }
+      }
+
+      fragmentManager
+          .beginTransaction()
+          .add(fragment, FRAGMENT_TAG)
+          .commitNowAllowingStateLoss()
```

Comment:
No, I don't want to care about fragments. If we change this to `commit()`, it'll lead us to a situation where we start seeing `IllegalStateExceptions` because there was a race condition between showing the dialog and stopping the activity.

As an alternative, we can mark the dialog as shown only once its canceled. 

That said, I don't see a strong reason for doing it. The Activity can only stop if the app gets killed midway. What other situations can there be?

Reply:
Yeah, makes sense. I was probably overthinking this anyway.